src/main/java/uk/gov/nationalarchives/pdi/step/xml/CanonicalizationResult.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.nationalarchives.pdi.step.xml;
+
+public class CanonicalizationResult {
+
+    private final String canonicalXml;
+    private String errorMessage = "";
+    private Boolean hasErrors = false;
```

Comment:
Does `hasErrors` need to be Tri-State?

Reply:
Not sure I understand. It doesn't have an error until it has en error.

Reply:
So `Boolean` is tri-state because it is an object, and that object can be one of two values. i.e. in total it can be either: `null`, `true`, or `false`.

Perhaps you only need the primitive `boolean` instead which is bi-state (or binary).

Reply:
Ah yes. I have to remember this is Java. Changed to `boolean`.